singlefactorauth/src/main/java/com/web3auth/singlefactorauth/SingleFactorAuth.kt

```diff
@@ -42,9 +45,25 @@ class SingleFactorAuth(
         sessionManager = SessionManager(ctx, sfaParams.getSessionTime(), ctx.packageName)
     }
 
-    fun initialize(ctx: Context): SFAKey {
-        val data = sessionManager.authorizeSession(ctx.packageName, ctx).get()
-        return gson.fromJson(JSONObject(data).toString(), SFAKey::class.java)
+    fun initialize(ctx: Context): CompletableFuture<SFAKey> {
+        val sfaCF = CompletableFuture<SFAKey>()
+        val data = sessionManager.authorizeSession(ctx.packageName, ctx)
+        data.whenComplete { response, error ->
+            val mainHandler = Handler(Looper.getMainLooper())
+            mainHandler.post {
+                if (error != null) {
+                    sfaCF.completeExceptionally(error)
+                } else {
+                    sfaCF.complete(
+                        gson.fromJson(
+                            JSONObject(response).toString(),
+                            SFAKey::class.java
+                        )
+                    )
+                }
+            }
+        }
+        return sfaCF
```

Comment:
Why did you do this?

Reply:
Done some changes in session-manager-sdk and this sdk. We can discuss those. This is done to return the correct error from initialize function.

Reply:
We had already discussed the problem with doing it this way and that it is not a substitute for proper error handling..